apps/dotcdn/src/app/dotcdn.component.store.ts

```diff
@@ -0,0 +1,214 @@
+
+                            this.addChartData(chartData);
+                            this.addStatsData(statsData);
+                        },
+                        (error) => {
+                            // TODO: Handle error
+                            console.log(error);
+                        }
+                    )
+                );
+            })
+        );
+    });
+
+    private setChartStatsData({ stats }: DotCDNStats) {
+        const chartData: ChartData = {
+            labels: this.getLabels(stats.bandwidthUsedChart),
+            datasets: [
+                {
+                    label: 'Bandwidth Used',
+                    data: Object.values(stats.bandwidthUsedChart),
+                    borderColor: '#42A5F5',
+                    fill: false
+                }
+            ]
+        };
+
+        const statsData: DotChartStats[] = [
+            {
+                label: 'Bandwidth Used',
+                value: stats.bandwidthPretty,
+                icon: 'insert_chart_outlined'
+            },
+            {
+                label: 'Requests Served',
+                value: `${stats.totalRequestsServed}`,
+                icon: 'file_download'
+            },
+            {
+                label: 'Cache Hit Rate',
+                value: `${stats.cacheHitRate.toFixed(2)}%`,
+                icon: 'file_download'
+            }
+        ];
+
+        return { chartData, statsData };
+    }
+
+    /**
+     * Purges the CDN cache
+     *
+     * @param {boolean} [invalidateAll=false]
+     * @param {string[]} [urls]
+     * @return {*}  {(Observable<ResponseView<any>> | void)}
+     * @memberof DotCDNStore
+     */
+    purgeCDNCache(invalidateAll: boolean): void;
+    purgeCDNCache(invalidateAll: boolean, urls?: string[]): Observable<ResponseView<any>>;
+    purgeCDNCache(
+        invalidateAll: boolean = false,
+        urls?: string[]
+    ): Observable<ResponseView<any>> | void {
+        if (!invalidateAll) {
+            this.dispatchLoading({
+                loadingState: LoadingState.LOADING,
+                loader: Loader.PURGE_URLS
+            });
+            return this.dotCdnService.purgeCache(invalidateAll, urls).pipe(
+                tap(() => {
+                    this.dispatchLoading({
+                        loadingState: LoadingState.LOADED,
+                        loader: Loader.PURGE_URLS
+                    });
+                })
+            );
+        }
+
+        this.dispatchLoading({
+            loadingState: LoadingState.LOADING,
+            loader: Loader.PURGE_PULL_ZONE
+        });
+        this.dotCdnService
+            .purgeCache(invalidateAll)
+            .pipe(take(1))
+            .subscribe(() => {
+                this.dispatchLoading({
+                    loadingState: LoadingState.LOADED,
+                    loader: Loader.PURGE_PULL_ZONE
+                });
+            });
+    }
```

Comment:
Don't use params as flag is better to break this method into 2, one to all and one to the list, using overload here makes it more complex than just break it.